### PR TITLE
feat: add support for RTK operations

### PR DESCRIPTION
**Pull request by doudou**
Depends on:
- [ ] https://github.com/tidewise/drivers-gps_ublox/pull/9

### CODE REVIEW

**Comment by doudou** [comment]
Ping

File: tasks/Task.cpp [hunk 1]
```diff
@@ -112,12 +115,12 @@ base::samples::RigidBodyState Task::convertToRBS(const gps_ublox::PVT &data) con
...
 
     auto geodetic = convertToBaseSolution(data);
-    base::samples::RigidBodyState nwu = mUTMConverter.convertToNWU(geodetic);
+    base::samples::RigidBodyState nwu = utmConverter.convertToNWU(geodetic);
     rbs.position = nwu.position;
     rbs.cov_position = nwu.cov_position;
     return rbs;
 }
-gps_base::SatelliteInfo Task::convertToBaseSatelliteInfo(const SatelliteInfo &sat_info) const
+static gps_base::SatelliteInfo convertToBaseSatelliteInfo(const SatelliteInfo &sat_info)
```
Comment (rodrigoleonello) [inline tasks/Task.cpp:138]: Why it is needed to convert to `static`methods?

File: gps_ublox.orogen [hunk 2]
```diff
@@ -39,6 +40,20 @@ task_context "Task", subclasses: "iodrivers_base::Task" do
     # Navigation solution's configuration
     property "navigation_configuration", "gps_ublox/configuration/Navigation"
 
+    # RTCM output configuration
```
Comment (rodrigoleonello) [inline gps_ublox.orogen:43]: what RTCM means?

File: tasks/Task.cpp (hunk 1 above)
Comment (caioaamaral) [reply tasks/Task.cpp:138]: they are out of the class scope now (because they are just helpers, I believe). So, being a free function it has to be static.

Comment (doudou) [reply tasks/Task.cpp:138]: They're not methods anymore, they're functions. They are not needed at the class level (they don't modify anything from the class), and it made it simpler to access within the PollCallbacks class

File: gps_ublox.orogen (hunk 2 above)
Comment (doudou) [reply gps_ublox.orogen:43]: https://gssc.esa.int/navipedia/index.php/RTK_Standards

It is a standard message format to transfer GNSS satellite observations (that is, a representation of the raw signal readings) between GNSS units

**Review by rodrigoleonello — APPROVED** [review]

File: tasks/Task.cpp [hunk 3]
```diff
@@ -112,12 +130,12 @@ base::samples::RigidBodyState Task::convertToRBS(const gps_ublox::PVT &data) con
         M_PI, Eigen::Vector3d::UnitX()) * may_invalidate(body2ned_velocity);
 
     auto geodetic = convertToBaseSolution(data);
-    base::samples::RigidBodyState nwu = mUTMConverter.convertToNWU(geodetic);
+    base::samples::RigidBodyState nwu = utmConverter.convertToNWU(geodetic);
```
Comment (caioaamaral) [inline tasks/Task.cpp:133]: nice cleanup

File: tasks/Task.cpp [hunk 4]
```diff
@@ -156,35 +174,73 @@ gps_base::Solution Task::convertToBaseSolution(const PVT &data) const
...
 }
-void Task::processIO()
-{
-    gps_ublox::UBX::Frame frame = mDriver->readFrame();
-    if (frame.msg_class == UBX::MSG_CLASS_NAV && frame.msg_id == UBX::MSG_ID_PVT) {
-        gps_ublox::PVT data = UBX::parsePVT(frame.payload);
 
-        _pose_samples.write(convertToRBS(data));
-        _gps_solution.write(convertToBaseSolution(data));
+struct gps_ublox::PollCallbacks : gps_ublox::Driver::PollCallbacks {
```
Comment (caioaamaral) [inline tasks/Task.cpp:197]: I wasn't aware that we could use a qualified name on struct definition

**Review by caioaamaral — APPROVED** [review]
All good!

File: tasks/Task.hpp [hunk 5]
```diff
@@ -26,11 +27,16 @@ namespace gps_ublox{
...
-	friend class TaskBase;
+        friend class TaskBase;
+        friend struct PollCallbacks;
+
     protected:
         gps_base::UTMConverter mUTMConverter;
         std::unique_ptr<gps_ublox::Driver> mDriver;
 
+        bool mOutputRTK;
+        RTKInfo mRTKInfo;
```
Comment (caioaamaral) [inline tasks/Task.hpp:38]: where are we including RTKInfo?

File: tasks/Task.cpp (hunk 1 above)
Comment (doudou) [reply tasks/Task.cpp:138]: > So, being a free function it has to be static.

"has to" is pretty strong here. It is better, sure, but it does not "have to".

File: tasks/Task.hpp (hunk 5 above)
Comment (doudou) [reply tasks/Task.hpp:38]: It is available because the task has a port of type RTKInfo

File: tasks/Task.cpp (hunk 4 above)
Comment (doudou) [reply tasks/Task.cpp:197]: Neither was I ... just did it without thinking about it.

File: tasks/Task.cpp (hunk 1 above)
Comment (caioaamaral) [reply tasks/Task.cpp:138]: > "has to" is pretty strong here. It is better, sure, but it does not "have to".

Indeed. Sometimes it saves you from some linkage problems

**Pull request merged**